src/sentry/preprod/api/endpoints/organization_preprod_artifact_assemble.py

```diff
@@ -0,0 +1,118 @@
+    publish_status = {
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+    permission_classes = (ProjectReleasePermission,)
+
+    def post(self, request: Request, project) -> Response:
+        """
+        Assembles a preprod artifact (mobile build, etc.) and stores it in the database.
+        """
+        with sentry_sdk.start_span(op="preprod_artifact.assemble"):
+            schema = {
+                "type": "object",
+                "properties": {
+                    "checksum": {"type": "string", "pattern": "^[0-9a-f]{40}$"},
+                    "chunks": {
+                        "type": "array",
+                        "items": {"type": "string", "pattern": "^[0-9a-f]{40}$"},
+                    },
+                    # Optional metadata
+                    "git_sha": {"type": "string"},
+                    "build_configuration": {"type": "string"},
+                },
+                "required": ["checksum", "chunks"],
+                "additionalProperties": False,
+            }
+
+            error_messages = {
+                "checksum": "The checksum field is required and must be a 40-character hexadecimal string.",
+                "chunks": "The chunks field is required and must be provided as an array of 40-character hexadecimal strings.",
+                "git_sha": "The git_sha field must be a string.",
+                "build_configuration": "The build_configuration field must be a string.",
+            }
+
+            try:
+                data = orjson.loads(request.body)
+                jsonschema.validate(data, schema)
+            except jsonschema.ValidationError as e:
+                error_message = e.message
+                # Get the field from the path if available
+                if e.path:
+                    if field := e.path[0]:
+                        error_message = error_messages.get(str(field), error_message)
+
+                return Response({"error": error_message}, status=400)
+            except Exception:
+                return Response({"error": "Invalid json body"}, status=400)
+
+            checksum = data.get("checksum")
+            chunks = data.get("chunks", [])
+
+            # Check if all requested chunks have been uploaded
+            missing_chunks = find_missing_chunks(project.organization_id, set(chunks))
+            if missing_chunks:
+                return Response(
+                    {
+                        "state": ChunkFileState.NOT_FOUND,
+                        "missingChunks": missing_chunks,
+                    }
+                )
+
+            # Check current assembly status
+            state, detail = get_assemble_status(AssembleTask.PREPROD_ARTIFACT, project.id, checksum)
+            if state == ChunkFileState.OK:
+                return Response({"state": state, "detail": detail, "missingChunks": []})
+            elif state is not None:
+                return Response({"state": state, "detail": detail, "missingChunks": []})
```

Comment:
The `if` and `elif` branches are the same, so this should be equivalent to
```suggestion
            if state is not None:
                return Response({"state": state, "detail": detail, "missingChunks": []})
```